src/contexts/auth.tsx

```diff
@@ -0,0 +1,92 @@
+import { PROVIDERS } from '@distributedlab/w3p'
+import { createContext, ReactNode, useCallback, useMemo, useState } from 'react'
+
+import { useMetamaskZkpSnapContext, useWeb3Context } from '@/hooks'
+import { authStore, useAuthState } from '@/store'
+
+interface AuthContextValue {
+  isAuthorized: boolean
+  login: () => Promise<void>
+  authorize: () => Promise<void>
+  logOut: () => Promise<void>
+}
+
+export const AuthContext = createContext<AuthContextValue>({
+  isAuthorized: false,
+  login: () => Promise.resolve(),
+  authorize: () => Promise.resolve(),
+  logOut: () => Promise.resolve(),
+})
+
+export const AuthContextProvider = ({ children }: { children: ReactNode }) => {
+  const { jwt: storeJwt } = useAuthState()
+  const { init, provider } = useWeb3Context()
+  const { isSnapInstalled, connectOrInstallSnap, checkSnapStatus } = useMetamaskZkpSnapContext()
+  const [isJwtValid, setIsJwtValid] = useState(false)
+
+  const isAuthorized = useMemo(
+    () => (provider?.isConnected && isSnapInstalled && isJwtValid) || false,
+    [isJwtValid, isSnapInstalled, provider?.isConnected],
+  )
+
+  const _setJwt = useCallback((jwt: string) => {
+    authStore.setJwt(jwt)
+  }, [])
```

Comment:
IMO we don't need this wrapper and useCallback at all

we can use `authStore.setJwt` directly as it's defined outside of the React scope